internal/ignition/ignition.go

```diff
@@ -1513,3 +1524,67 @@ func proxySettingsForIgnition(httpProxy, httpsProxy, noProxy string) (string, er
 	}
 	return buf.String(), nil
 }
+
+func getIcspFileFromInstallConfig(cfg []byte) (string, error) {
+	contents, err := getIcsp(cfg)
+	if err != nil {
+		return "", err
+	}
+	if contents == nil {
+		return "", nil
+	}
+
+	icspFile, err := ioutil.TempFile("", "icsp-file")
+	if err != nil {
+		return "", err
+	}
+	if _, err := icspFile.Write(contents); err != nil {
+		os.Remove(icspFile.Name())
```

Comment:
I would defer this as well, just so it executes after the file is already closed.

Reply:
Changed to add the Close if the Write fails, seems more readable than using defer